wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -183,43 +187,66 @@ class WirePlugin @Inject constructor(
     }
   }
 
+
   private fun mergeDependencyPaths(
     project: Project,
     dependencyPaths: Set<String>,
-    dependencyTrees: Set<SourceDirectorySet> = emptySet()
+    dependencyTrees: Set<SourceDirectorySet> = emptySet(),
+    dependencyJars: Set<ProtoRootSet> = emptySet()
   ): List<Any> {
     val allPaths = dependencyTrees.toMutableList<Any>()
 
     val parser = FileOrUriNotationConverter.parser()
+
+    dependencyJars.forEach { depJar ->
+      depJar.srcJar?.let { path ->
+        allPaths += mapDependencyPath(parser, path, project, depJar.includes)
+      }
+    }
+
     dependencyPaths.forEach { path ->
-      val converted = parser.parseNotation(path)
+      allPaths += mapDependencyPath(parser, path, project)
+    }
 
-      if (converted is File) {
-        val file =
-          if (!converted.isAbsolute) File(project.projectDir, converted.path) else converted
-        if (!file.exists()) {
-          throw IllegalArgumentException(
-              "Invalid path string: \"$path\". Path does not exist."
-          )
-        }
-        if (file.isDirectory) {
-          allPaths += project.files(path) as Any
-        } else {
-          throw IllegalArgumentException(
-              "Invalid path string: \"$path\". For individual files, use the closure syntax."
-          )
-        }
-      } else if (converted is URI && isURL(converted)) {
+    return allPaths
+  }
+
+  private fun mapDependencyPath(
+    parser: NotationParser<Any, Any>,
+    path: String,
+    project: Project,
+    jarIncludes: List<String> = emptyList()
+  ): Any {
+    val converted = parser.parseNotation(path)
+
+    if (converted is File) {
+      val file =
+        if (!converted.isAbsolute) File(project.projectDir, converted.path) else converted
+      if (!file.exists()) {
         throw IllegalArgumentException(
-            "Invalid path string: \"$path\". URL dependencies are not allowed."
+            "Invalid path string: \"$path\". Path does not exist."
+        )
+      }
+      return when {
+        file.isDirectory -> project.files(path)
+        file.isJar -> {
+          if (jarIncludes.isNotEmpty()) {
+            jarToIncludes[file.path] = jarIncludes
```

Comment:
I find this side-effect surprising

Reply:
Not sure if I see a better way though.

Still I don’t like how this method returns a value and changes local state, and the 2nd part is kind of invisible.

Reply:
Return a Pair/custom class and let the caller process it as needed?

Reply:
Agreed on all points.  Can revisit down the road, but this unblocks in the short term.

Reply:
@Egorand here's one way that can look: 
https://github.com/square/wire/commit/4e4351306416c7df90db119c6e6eb4d2a01aae1c

not sure i like it better though.

Reply:
spoke with @Egorand offline who gave some suggestions using sealed classes.  i'll merge this and follow up.